app/models/file_attachment/revision.rb

```diff
@@ -12,7 +12,17 @@ class FileAttachment::Revision < ApplicationRecord
                           foreign_key: "file_attachment_revision_id",
                           join_table: "revisions_file_attachment_revisions"
 
-  delegate :title, to: :metadata_revision
+  delegate :title,
+           :isbn,
+           :unique_reference,
+           :paper_number,
+           :unofficial?,
+           :command_paper?,
+           :act_paper?,
```

Comment:
Could you help me level up my Rails knowledge and explain how `act_paper?`, `command_paper?` and `unofficial?` work?

I know Rails creates some helper methods automatically based on a model's fields, but how does it know to look under the `official_document` property for these? What would happen if we had defined `enum official_document: { isbn: "foo" }` - would that override the `isbn?` helper method?

Reply:
Sure, it is quite surprising. I copied it from our approach to revision state e.g. `revision.major?`. Docs for it seem to be pretty comprehensive...

https://api.rubyonrails.org/v5.2.4.1/classes/ActiveRecord/Enum.html